### PR TITLE
Make Silk.NET buildable on Linux by skipping iOS and Mac Catalyst stuff.

### LINKED ISSUE #2014: Make Silk.NET buildable on Linux (by loosening .NET workload requirements)
Silk isn't currently buildable on Linux due to the `ios` and `maccatalyst` workload requirements. These are not installable on Linux (despite `macos` being!). Per discussion on Discord, the build system should exclude the corresponding TFMs if building on Linux.

Mostly logging this issue as a reminder to myself to open a PR to fix this.

### PR DESCRIPTION
**Pull request by alexrp**
Also update build instructions.

Closes #2014.

### CODE REVIEW

**Review by Perksey — APPROVED** [review]
Did I review the wrong PR before?

File: README.md [hunk 1]
```diff
@@ -74,16 +74,18 @@ In addition, the Silk.NET working group help drive larger user-facing changes pr
...
-- **Should**: Android SDK version 30 with NDK tools installed. On Windows, for best results this should be installed into `C:\ProgramData\Android\android-sdk`.
-- **Could**: Java JDK (for gradle)
-- **Could**: Visual Studio 2022 Community version 17.0 or later
+- .NET 7 SDK
+- Android, iOS, and Mac Catalyst workloads (use `dotnet workload install android ios maccatalyst` to install them)
+    - On Linux, `ios` and `maccatalyst` should be omitted as they are not available
+- Android SDK version 30 with NDK tools installed
+    - On Windows, for best results this should be installed into `C:\ProgramData\Android\android-sdk`
+- Java 11 JDK
+- Visual Studio 2022 Community version 17.0 or later (optional)
```
Comment (Perksey) [inline README.md:83]: Why did you remove the must/should/could?

Comment (alexrp) [reply README.md:83]: Because the only one that's actually "could" now is Visual Studio. All the other stuff is mandatory to do a successful `./build.sh`.

Comment (Perksey) [reply README.md:83]: ok

**Review by Perksey — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -74,16 +74,18 @@ In addition, the Silk.NET working group help drive larger user-facing changes pr
 <h1 align="center">Building from source</h1>
 
 Prerequisites
-- **Must**: .NET 6 SDK
```
[the remark this answers is not in the document]
Comment (alexrp) [reply README.md:77]: I think you're right. I forgot I also had .NET 6 installed, so it just worked. Updated it to mention both.

**Review by Beyley — APPROVED** [review]
it would be cool if linux builds could be tested in CI aswell, but thats something that can be done at a later time by whoever wants to take that on

**Comment by alexrp** [comment]
Seems like it shouldn't be too hard to add a Linux leg since the current CI is macOS-based. I can take a look at that after I finish my current work on native library builds.
